### PR TITLE
configure: now supports the --without-stream_ssl_module option to

### LINKED ISSUE #388: can't disable openssl
version:  openresty-1.13.6.2.tar.gz  

run:     ./configure --prefix=/data/qqpet/openresty/ --without-http_ssl_module  --without-http_rewrite_module

### PR DESCRIPTION
**Pull request by dndx**
disable ngx_stream_ssl_module if necessary. fixes #388.

I hereby granted the copyright of the changes in this pull request
to the authors of this openresty project.

### CODE REVIEW

File: util/configure [hunk 1]
```diff
@@ -488,7 +495,17 @@ _END_
...
-        push @ngx_opts, '--with-stream', '--with-stream_ssl_module';
+        push @ngx_opts, '--with-stream';
+    }
+
+    if ($opts->{no_stream_ssl} && $opts->{http_stream}) {
+        die "--with-stream_ssl_module conflicts with --without-stream_ssl_module.",
+            "\n";
+    }
+
+    if (! $opts->{no_stream_ssl} && ! $opts->{stream_ssl}) {
```
Comment (agentzh) [inline util/configure]: Better avoid adding space after `!` for the sake of consistency.

File: util/configure [hunk 2]
```diff
@@ -1234,6 +1251,7 @@ _EOC_
     $msg .= <<'_EOC_';
   --without-ngx_devel_kit_module     disable ngx_devel_kit_module
   --without-http_ssl_module          disable ngx_http_ssl_module
+  --without-stream_ssl_module        disable ngx_stream_ssl_module
 
 _EOC_
```
Comment (agentzh) [inline util/configure]: Will you add some tests to the existing test suite to cover these changes? The `t/sanity.t` file is the right place to add such tests.

**Comment by dndx** [comment]
@agentzh Addressed.

**Comment by agentzh** [comment]
Merged with minor test fixes.

**Pull request closed without merging**